app/models/job_application.rb

```diff
@@ -1,5 +1,5 @@
 class JobApplication < ApplicationRecord
-  before_save :update_status_timestamp, if: :will_save_change_to_status?
+  before_save :update_status_timestamp, if: %i[will_save_change_to_status? ignore_for_offered_and_declined?]
```

Comment:
It took me a while to understand the purpose of `ignore_for_offered_and_declined?`.
So maybe we could add a comment here explaining that these two statuses' timestamps are set by manual input recording through forms instead of the automated timestamp. 
Thinking of someone stumbling into this bit of code with even less context than I (not having reviewed the PR introducing all this, but just landing on this file/behaviour).